src/utils/HealthLogger.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, jQuery*/
+
+/**
+ *  Utilities functions related to Health data loggig
+ */
+define(function (require, exports, module) {
+    "use strict";
+
+    var PreferencesManager          = require("preferences/PreferencesManager"),
+        LanguageManager             = require("language/LanguageManager"),
+        FileUtils                   = require("file/FileUtils"),
+        PerfUtils                   = require("utils/PerfUtils"),
+
+        HEALTH_DATA_STATE_KEY       = "HealthData.Logs",
+        logHealthData               = true;
+
+    /**
+     * Init: creates the health log preference keys in the state.json file
+     */
+    function init() {
+        PreferencesManager.stateManager.definePreference(HEALTH_DATA_STATE_KEY, "object", {});
+    }
+
+    /**
+     * Return the Performance related data
+     * @returns {Object} Performance Data aggregated till now
+     */
+    function getPerformanceData() {
+        var perfData = PerfUtils.getHealthReport();
+        return perfData;
+    }
+
+    /**
+     * Return the aggregate of all health data logged till now from all sources
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getAggregatedHealthData() {
+        var healthData = PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+        jQuery.extend(healthData, getPerformanceData());
+        return healthData;
+    }
+
+    /**
+     * Return all health data logged till now stored in the state prefs
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getHealthData() {
+        var healthData = PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+        return healthData;
+    }
```

Comment:
[This](https://github.com/adobe/brackets/pull/11311/files#r33618816) could also apply here.